In [ ]:
#install pingouin package
%pip install pingouin

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 204.4/204.4 kB 4.6 MB/s eta 0:00:00


In [ ]:
#import modules
import math
import statistics
import numpy as np
import scipy.stats
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
import statsmodels.api as sm
from sklearn.preprocessing import StandardScaler
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
import pylab as py
import os
os.environ['KMP_DUPLICATE_LIB_OK']='True'

In [ ]:
df = pd.read_csv('merged_data.csv')
df.head(5)

Sl.No  Age  Physical Interpretation  Social Interpretation.1  \
0      1   21        31  Above average      39             High   
1      2   18        34           High      38             High   
2      3   21        33           High      21          Average   
3      4   19        29  Above average      29    Above average   
4      5   18        26  Above average      25    Above average   

   Temperamental Interpretation.2  Educational Interpretation.3  Moral  \
0             28    Above average           27    Above average     28   
1             28    Above average           32    Above average     34   
2             22          Average           27    Above average     34   
3             30    Above average           30    Above average     32   
4             27    Above average           23          Average     25   

  Interpretation.4  Intellectual Interpretation.5  TOTAL Interpretation.6  \
0    Above average            22          Average    175    Above Average   
1             High            37             High    199             High   
2             High            27    Above average    164    Above average   
3    Above average            24          Average    174    Above average   
4    Above average            24          Average    150    Above average   

  Gender  TOTAL_BAI Interpretaion_BAI  
0      M          8          Very low  
1      M         14          Very low  
2      M         26          Moderate  
3      M          6          Very low  
4      M         13          Very low

## H1: Self-concept dimensions (i.e., physical, social, temperamental, educational, moral, and intellectual) increases the impact of anxiety.

To determine the relationship between self-concept and anxiety, the study tested six parameters to predict their impact on
anxiety. The tested parameters were: Physical, Social, Temperamental, Educational, Moral and intellectual. The correlation
coefficient, R, was calculated and used to predict the relationship between self-concept and anxiety

In [ ]:
import numpy as np
import pandas as pd
from sklearn.metrics import mean_squared_error
import statsmodels.api as sm


# Independent variables (predictors)
X = df[['Intellectual', 'Moral', 'Educational', 'Social', 'Physical', 'Temperamental']]

# Dependent variable
y = df['TOTAL_BAI']

# Add constant for statsmodels
X_with_const = sm.add_constant(X)

# Fit the model
model = sm.OLS(y, X_with_const).fit()

# Print the model summary
print(model.summary())

# Extract R, R^2, Adjusted R^2, and Std. Error of the Estimate
r_square = model.rsquared
adjusted_r_square = model.rsquared_adj

# Number of observations and predictors
n = len(y)
p = X.shape[1]
# Calculate Residual Sum of Squares (RSS)
rss = np.sum((y - model.fittedvalues) ** 2)

# Calculate Std. Error
std_error_rss = np.sqrt(rss / (n - p - 1))
r = np.sqrt(r_square)

print("==============================================================================")

print(f"\nR: {r}")
print(f"R^2: {r_square}")
print(f"Adjusted R^2: {adjusted_r_square}")
print(f"Std. Error of the Estimate:{std_error_rss}")

                            OLS Regression Results                            
Dep. Variable:              TOTAL_BAI   R-squared:                       0.211
Model:                            OLS   Adj. R-squared:                  0.188
Method:                 Least Squares   F-statistic:                     9.088
Date:                Sat, 30 Nov 2024   Prob (F-statistic):           8.23e-09
Time:                        21:00:01   Log-Likelihood:                -829.28
No. Observations:                 211   AIC:                             1673.
Df Residuals:                     204   BIC:                             1696.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const            55.3233      9.015      6.137

A multiple regression analysis was conducted to examine the relationship between several predictor variables (Intellectual, Moral, Educational, Social, Physical, and Temperamental) and the outcome variable, TOTAL_BAI. The model was statistically significant, *F*(6, 204) = 9.09, *p* < .001, accounting for 21.1% of the variance in TOTAL_BAI.

Of the predictor variables, only Physical and Temperamental had a significant negative relationship with TOTAL_BAI, *β* = -.84, *p* < .001, and *β* = -.67, *p* < .05, respectively. The remaining predictors were not significantly related to the outcome.

In [ ]:
# Regression ANOVA without SCaling

# Define predictors and dependent variable
X = df[["Physical", "Social", "Temperamental", "Educational", "Moral", "Intellectual"]]
y = df["TOTAL_BAI"]

# Add constant to the model
X = sm.add_constant(X)

# Fit the OLS model
model = sm.OLS(y, X).fit()

# Get Regression Sum of Squares (SSR), Residual Sum of Squares (SSE), and Total Sum of Squares (SST)
sst = ((y - y.mean()) ** 2).sum()  # Total sum of squares
ssr = model.ess  # Regression sum of squares
sse = model.ssr  # Residual sum of squares

# Degrees of freedom
df_model = model.df_model  # Number of predictors
df_resid = model.df_resid  # Residual degrees of freedom
df_total = len(y) - 1  # Total degrees of freedom

# Mean Squares
msr = ssr / df_model  # Mean square for regression
mse = sse / df_resid  # Mean square for residuals

# F-statistic
f_stat = msr / mse

# p-value for F-statistic
p_value = model.f_pvalue

# Display ANOVA table
anova_table = pd.DataFrame({
    "Source": ["Regression", "Residual", "Total"],
    "Sum of Squares": [ssr, sse, sst],
    "df": [df_model, df_resid, df_total],
    "Mean Square": [msr, mse, None],
    "F": [f_stat, None, None],
    "Sig.": [p_value, None, None]
})

print("ANOVA Table:")
print(anova_table.round(3))

ANOVA Table:
       Source  Sum of Squares     df  Mean Square      F  Sig.
0  Regression        8561.792    6.0     1426.965  9.088   0.0
1    Residual       32032.113  204.0      157.020    NaN   NaN
2       Total       40593.905  210.0          NaN    NaN   NaN


The effect of the six self-concept dimensions (Intellectual, Educational, Moral, Social, Physical, and Temperamental) on anxiety (BAI). The results indicated a significant effect of the self-concept dimensions on anxiety, F(6, 204) = 9.088, p < .001.

## 3.1 Estimated model Coefficients

In [ ]:
import pandas as pd
import statsmodels.api as sm
from sklearn.preprocessing import StandardScaler



# Define predictors and dependent variable
X = df[['Physical', 'Social', 'Temperamental', 'Educational', 'Moral', 'Intellectual']]
y = df['TOTAL_BAI']

# Add constant to the model
X_with_const = sm.add_constant(X)

# Fit the OLS model
model = sm.OLS(y, X_with_const).fit()

# Standardize the independent variables (Z-score normalization)
scaler = StandardScaler()
X_standardized = pd.DataFrame(scaler.fit_transform(X), columns=X.columns)
X_standardized_with_const = sm.add_constant(X_standardized)

# Fit the OLS model for standardized coefficients
model_standardized = sm.OLS(y, X_standardized_with_const).fit()

# Extract results for unstandardized and standardized coefficients
coefficients = pd.DataFrame({
    'Unstandardised Coefficients (B)': model.params,
    'Standardised Coefficients (Beta)': model_standardized.params[1:],  # Skip constant for standardized
    'Std. Error': model.bse,
    't': model.tvalues,
    'Sig.': model.pvalues
})

# Print the table
print(coefficients.round(3))

               Unstandardised Coefficients (B)  \
Educational                             -0.340   
Intellectual                             0.527   
Moral                                   -0.107   
Physical                                -0.841   
Social                                   0.254   
Temperamental                           -0.674   
const                                   55.323   

               Standardised Coefficients (Beta)  Std. Error      t   Sig.  
Educational                              -1.456       0.239 -1.419  0.157  
Intellectual                              1.832       0.279  1.886  0.061  
Moral                                    -0.412       0.269 -0.399  0.690  
Physical                                 -3.982       0.233 -3.606  0.000  
Social                                    1.079       0.232  1.095  0.275  
Temperamental                            -2.918       0.277 -2.436  0.016  
const                                       NaN       9.015  6.137 

An unstandardized coefficient indicates how much the dependent variable varies with an independent variable.

Among the predictors, Physical and Temperamental were statistically significant predictors of the dependent variable (p < .05 for both), demonstrating strong negative relationships. The predictors Educational, Intellectual, and Social had effects that were not statistically significant (p > .05 for all), indicating weaker or inconsistent relationships. The constant term provided a reliable baseline value for the dependent variable when all predictors were zero. These findings emphasize the importance of Physical and Temperamental in predicting the dependent variable, while other predictors may require further investigation or could be excluded from the model due to their lack of significance.

##  H2: There is a significant mean difference between the physical, social, Temperamental, Educational, Intellectual, and Moral self-concepts of males and females. - Group Statistics -

In [ ]:
import pandas as pd
import numpy as np

# Group the DataFrame by 'Gender'
grouped_df = df.groupby('Gender')

# List of columns to summarize
columns_to_summarize = ['Physical', 'Social', 'Temperamental', 'Educational', 'Intellectual', 'Moral', 'TOTAL_BAI']

# Initialize a list to store the results
results = []

# Loop through the columns and calculate statistics for each group
for col in columns_to_summarize:
    for gender, group in grouped_df:
        n = group[col].count()
        mean = group[col].mean()
        std_dev = group[col].std()
        std_error = std_dev / np.sqrt(n)
        results.append([col, gender, n, mean, std_dev, std_error])

# Create a DataFrame for the results
summary_table = pd.DataFrame(
    results,
    columns=['Variable', 'Gender', 'N', 'Mean', 'Std. Deviation', 'Std. Error Mean']
)

# Format the table while preserving the specified order
summary_table['Variable'] = pd.Categorical(summary_table['Variable'], categories=columns_to_summarize, ordered=True)
summary_table_sorted = summary_table.sort_values('Variable')

# Pivot the sorted DataFrame
summary_table_formatted = summary_table_sorted.pivot_table(
    index='Variable',
    columns='Gender',
    values=['N', 'Mean', 'Std. Deviation', 'Std. Error Mean']
)

# Print the formatted table
print("Table 3.4: Group Statistics")
print(summary_table_formatted.to_string())

Table 3.4: Group Statistics
                    Mean                 N        Std. Deviation            Std. Error Mean          
Gender                 F          M      F      M              F          M               F         M
Variable                                                                                             
Physical       28.056075  28.384615  107.0  104.0       4.079012   5.363139        0.394333  0.525899
Social         29.074766  28.548077  107.0  104.0       3.768541   4.706698        0.364319  0.461530
Temperamental  28.037383  28.144231  107.0  104.0       3.877670   4.791615        0.374869  0.469857
Educational    27.196262  26.567308  107.0  104.0       4.124111   4.467280        0.398693  0.438053
Intellectual   25.747664  26.375000  107.0  104.0       3.099280   3.829339        0.299619  0.375498
Moral          30.214953  29.730769  107.0  104.0       3.389852   4.279622        0.327709  0.419651
TOTAL_BAI      22.308411  20.432692  107.0  104.0     

<ipython-input-8-0f7a7439afd6>:33: FutureWarning: The default value of observed=False is deprecated and will change to observed=True in a future version of pandas. Specify observed=False to silence this warning and retain the current behavior
  summary_table_formatted = summary_table_sorted.pivot_table(


A **descriptive analysis** was conducted to compare the self-concept dimensions (physical, social, temperamental, educational, intellectual, and moral) between males and females. The results are as follows:

1. **Physical Self-Concept**:
   - Females (M = 28.06, SD = 4.08) reported slightly lower scores than males (M = 28.38, SD = 5.36). The standard error of the mean was 0.39 for females and 0.53 for males, suggesting comparable variability.

2. **Social Self-Concept**:
   - Females (M = 29.07, SD = 3.77) scored slightly higher than males (M = 28.55, SD = 4.71), with standard errors of 0.36 (females) and 0.46 (males).

3. **Temperamental Self-Concept**:
   - The mean scores for females (M = 28.04, SD = 3.88) and males (M = 28.14, SD = 4.79) were nearly identical, with standard errors of 0.37 and 0.47, respectively.

4. **Educational Self-Concept**:
   - Females (M = 27.20, SD = 4.12) scored higher than males (M = 26.57, SD = 4.47). The standard error was 0.40 for females and 0.44 for males, indicating consistent variability.

5. **Intellectual Self-Concept**:
   - Females (M = 25.75, SD = 3.10) reported slightly lower scores than males (M = 26.38, SD = 3.83). Standard errors were 0.30 and 0.38 for females and males, respectively.

6. **Moral Self-Concept**:
   - Females (M = 30.21, SD = 3.39) scored higher than males (M = 29.73, SD = 4.28), with standard errors of 0.33 (females) and 0.42 (males).

7. **TOTAL_BAI**:
   - Females (M = 22.31, SD = 14.35) scored higher than males (M = 20.43, SD = 13.43). Standard errors were 1.39 and 1.32 for females and males, respectively.

### Interpretation:
The descriptive statistics show small differences between males and females across self-concept dimensions. Females tended to report higher scores in **social**, **educational**, and **moral** self-concepts, while males scored slightly higher in **physical** and **intellectual** self-concepts. For **temperamental self-concept**, the means were almost identical.

## H3: There is a significant mean difference between the anxiety scores of males and females.

## Independent samples test

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as stats

# Load your data

# Map Gender values to 0 and 1 (assuming 1 for male, 2 for female)
df['Gender_num'] = df['Gender'].map({'M': 1, 'F': 2})

# Define the columns to test
variables = ['Physical', 'Social', 'Temperamental', 'Educational', 'Moral', 'Intellectual', 'TOTAL_BAI']

# Function to perform independent samples t-test
def perform_t_tests(df, variables):
    results = []
    for var in variables:
        group1 = df[df['Gender_num'] == 1][var]
        group2 = df[df['Gender_num'] == 2][var]

        # Levene's Test for Equality of Variances
        levene_stat, levene_p = stats.levene(group1, group2)

        # T-test for Equality of Means
        t_stat_eq_var, t_p_eq_var = stats.ttest_ind(group1, group2, equal_var=True)
        t_stat_uneq_var, t_p_uneq_var = stats.ttest_ind(group1, group2, equal_var=False)

        # Calculate mean difference and confidence intervals
        mean_diff = group1.mean() - group2.mean()

        # Equal variances assumed
        std_err_diff_eq_var = np.sqrt((group1.var() / len(group1)) + (group2.var() / len(group2)))
        df_eq_var = len(group1) + len(group2) - 2
        ci_lower_eq_var = mean_diff - 1.96 * std_err_diff_eq_var
        ci_upper_eq_var = mean_diff + 1.96 * std_err_diff_eq_var

        # Equal variances not assumed
        se_diff_uneq_var = np.sqrt((group1.var()/len(group1)) + (group2.var()/len(group2)))
        df_uneq_var = ((group1.var()/len(group1)) + (group2.var()/len(group2)))**2 / ((group1.var()**2/(len(group1)**2*(len(group1)-1))) + (group2.var()**2/(len(group2)**2*(len(group2)-1))))
        ci_lower_uneq_var = mean_diff - 1.96 * se_diff_uneq_var
        ci_upper_uneq_var = mean_diff + 1.96 * se_diff_uneq_var

        results.append({
            'Variable': var,
            'Levene\'s Test F': levene_stat,
            'Levene\'s Test p-value': levene_p,
            't-statistic (Equal Var)': t_stat_eq_var,
            'df (Equal Var)': df_eq_var,
            't-test p-value (Equal Var)': t_p_eq_var,
            'Mean Difference': mean_diff,
            'Std. Error Diff (Equal Var)': std_err_diff_eq_var,
            '95% CI Lower (Equal Var)': ci_lower_eq_var,
            '95% CI Upper (Equal Var)': ci_upper_eq_var,
            't-statistic (Unequal Var)': t_stat_uneq_var,
            'df (Unequal Var)': df_uneq_var,
            't-test p-value (Unequal Var)': t_p_uneq_var,
            'Std. Error Diff (Unequal Var)': se_diff_uneq_var,
            '95% CI Lower (Unequal Var)': ci_lower_uneq_var,
            '95% CI Upper (Unequal Var)': ci_upper_uneq_var
        })
    return pd.DataFrame(results)

# Perform the t-tests
results_df = perform_t_tests(df, variables)

# Display the results
print(results_df)

        Variable  Levene's Test F  Levene's Test p-value  \
0       Physical         3.847624               0.051144   
1         Social         4.251232               0.040459   
2  Temperamental         4.835174               0.028982   
3    Educational         0.543115               0.461971   
4          Moral         3.313412               0.070147   
5   Intellectual         3.603997               0.059019   
6      TOTAL_BAI         0.467301               0.494989   

   t-statistic (Equal Var)  df (Equal Var)  t-test p-value (Equal Var)  \
0                 0.501730             209                    0.616386   
1                -0.898542             209                    0.369931   
2                 0.178291             209                    0.858667   
3                -1.063054             209                    0.288984   
4                -0.912334             209                    0.362644   
5                 1.309798             209                    0.191702   
6

## H4: There is a relation between overall self-concept and anxiety - Correlation

In [ ]:
import numpy as np
from scipy.stats import pearsonr

# Example data (replace these arrays with your actual data)
self_concept = df['TOTAL']
bai = df['TOTAL_BAI']

# Calculate Pearson correlation
correlation, p_value = pearsonr(self_concept, bai)

print(f"correlation",{correlation})
print(f"p_value",{p_value})

correlation {-0.3536930906699036}
p_value {1.3003703089347565e-07}


A significant negative correlation was found between overall self-concept and anxiety (r = -.35, p < .001). This indicates that as overall self-concept increases, anxiety tends to decrease.